src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -161,69 +259,75 @@ def _create_osd(self, svc_arg=None, inbuf=None):
                 drive_group = orchestrator.DriveGroupSpec.from_json(json.loads(inbuf))
             except ValueError as e:
                 msg = 'Failed to read JSON input: {}'.format(str(e)) + usage
-                return HandleCommandResult(-errno.EINVAL, stderr=msg)
+                yield HandleCommandResult(-errno.EINVAL, stderr=msg)
+                return
 
         elif svc_arg:
             try:
                 node_name, block_device = svc_arg.split(":")
                 block_devices = block_device.split(',')
             except (TypeError, KeyError, ValueError):
                 msg = "Invalid host:device spec: '{}'".format(svc_arg) + usage
-                return HandleCommandResult(-errno.EINVAL, stderr=msg)
+                yield HandleCommandResult(-errno.EINVAL, stderr=msg)
+                return
 
             devs = orchestrator.DeviceSelection(paths=block_devices)
             drive_group = orchestrator.DriveGroupSpec(node_name, data_devices=devs)
         else:
-            return HandleCommandResult(-errno.EINVAL, stderr=usage)
+            yield HandleCommandResult(-errno.EINVAL, stderr=usage)
+            return
 
         # TODO: Remove this and make the orchestrator composable
         #   Like a future or so.
-        host_completion = self.get_hosts()
-        self._orchestrator_wait([host_completion])
+        host_completion = yield self.get_hosts()
         all_hosts = [h.name for h in host_completion.result]
 
         try:
             drive_group.validate(all_hosts)
         except orchestrator.DriveGroupValidationError as e:
-            return HandleCommandResult(-errno.EINVAL, stderr=str(e))
+            yield HandleCommandResult(-errno.EINVAL, stderr=str(e))
+            return
 
-        completion = self.create_osds(drive_group, all_hosts)
-        self._orchestrator_wait([completion])
+        completion = yield self.create_osds(drive_group, all_hosts)
```

Comment:
Maybe this should be part of other review/pr, but i think it is worth to comment here:
I think k that if we move : 
- <Args> to <drive group> object stuff
and
- hosts validation
Into each of the backend orchestrators , the code will be more clean  here.  

Look that this method is different from the rest.... here the orchestrator-cli is doing part of the things that the orchcestrators backends should do. If we do not change this... we are forcing to do the same in each client that will use the "osd create" method in no matter what backend orchestrator.